test/common/common.go

```diff
@@ -0,0 +1,446 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	googleDNSServer            = "8.8.8.8"
+	e2eSeparateOperandNsEnvVar = "E2E_SEPARATE_OPERAND_NAMESPACE"
+	operandNamespace           = "external-dns"
+	rbacRsrcName               = "external-dns-operator"
+	operatorServiceAccount     = "external-dns-operator"
+	OperatorNamespace          = "external-dns-operator"
+	dnsProviderEnvVar          = "DNS_PROVIDER"
+	e2eSkipDNSProvidersEnvVar  = "E2E_SKIP_DNS_PROVIDERS"
+	DnsPollingInterval         = 15 * time.Second
+	DnsPollingTimeout          = 3 * time.Minute
+)
+
+func DefaultService(name, namespace string) *corev1.Service {
+	return testService(name, namespace, corev1.ServiceTypeLoadBalancer)
+}
+
+func testService(name, namespace string, svcType corev1.ServiceType) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"external-dns.mydomain.org/publish": "yes",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"name": "hello-openshift",
+			},
+			Type: svcType,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+				},
+			},
+		},
+	}
+}
+
+func MustGetEnv(name string) string {
+	val := os.Getenv(name)
+	if val == "" {
+		panic(fmt.Sprintf("environment variable %s must be set", name))
+	}
+	return val
+}
+
+func RootCredentials(kubeClient client.Client, name string) (map[string][]byte, error) {
+	secret := &corev1.Secret{}
+	secretName := types.NamespacedName{
+		Name:      name,
+		Namespace: "kube-system",
+	}
+	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
+		return nil, fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
+	}
+	return secret.Data, nil
+}
+
+func LookupARecord(host, server string) ([]string, error) {
+	dnsClient := &dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	response, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Answer) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	var ips []string
+	for _, ans := range response.Answer {
+		if aRec, ok := ans.(*dns.A); ok {
+			ips = append(ips, aRec.A.String())
+		}
+	}
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	return ips, nil
+}
+
+// DnsQueryClusterInternal queries for a DNS hostname inside the VPC of a cluster using a dig pod. It returns a
+// map structure of the resolved IPs for easy lookup.
+func DnsQueryClusterInternal(ctx context.Context, t *testing.T, cl client.Client, cs *kubernetes.Clientset, namespace, hostname string) (map[string]struct{}, error) {
```

Comment:
```suggestion
func LookupARecordInternal(ctx context.Context, t *testing.T, cl client.Client, cs *kubernetes.Clientset, namespace, host string) ([]string, error) {
```
To make lookup functions look alike, WDYT?

Reply:
Yes, consistency is good. Done.